test/tap-filter

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+#
+# Tap parser that filters out test success and only prints the errors and
+# test summary.
+#
+# The parser keeps a buffer of all output during a test (including non-tap
+# data). If that test has a failing assertion, the entire buffer is printed.
+#
+# This filter adds one non-standard tap output line:
+#
+#    #: This is an example line that will be printed immediately
+#
+# You can use the "#:" prefix to a line and it will be passed through the
+# filter to stdout. This is using for printing status information.
+#
+declare test_name=
+declare test_error=false
+declare buffer=
+declare exit_code=0
+
+#
+# Add the passed-in data to the buffer.
+#
+_add_to_buffer() {
+	for data in "$@"; do
+		buffer="$buffer"$'\n'"$data"
+	done
+}
+
+while IFS= read -r line; do
+	case $line in
+		\#*)
+			# Beginning of new test...
+
+			# If there was an error during the last test, print the error
+			# before resetting
+			if $test_error; then
+				echo >&2
```

Comment:
Nitpick: I think it would be easier to read, and harder to miss, if we have an explicit echo-like function which prints to stdout. E.g.

```bash
_log() {
  echo "$@" >&2
}
```
... and us that instead of `>&2` after every echo.

I will not push hard for it, just think about it.